packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -221,10 +221,9 @@ class CupertinoSheetTransition extends StatefulWidget {
       reverseCurve: reverseCurve,
       parent: secondaryAnimation,
     );
-    final double deviceCornerRadius = MediaQuery.maybeViewPaddingOf(context)?.top ?? 0;
 
     final Animatable<BorderRadiusGeometry> decorationTween = Tween<BorderRadiusGeometry>(
-      begin: BorderRadius.circular(deviceCornerRadius),
+      begin: BorderRadius.circular(0),
```

Comment:
Ah, I was [looking into this issue too](https://github.com/flutter/flutter/issues/163334#issuecomment-2672298525).

This would fix phones with no rounded corners, but it would look not as smooth on phones with rounded corners. The "brute force" method I was thinking of was for the radius above, hard coding it to zero if the animation is at the beginning. This would likely be jumpy on screens without rounded corners, however, so it's not perfect either.

Reply:
Yaa, correct. When i saw animation on iPhone SE, in the beginning it jump for flat to 20 and then comes down to 12. And going back was smooth but it retained that curve.